src/test/kotlin/uk/gov/dluhc/printapi/messaging/ApplicationRemovedMessageListenerTest.kt

```diff
@@ -0,0 +1,44 @@
+package uk.gov.dluhc.printapi.messaging
+
+import org.assertj.core.api.Assertions.assertThat
+import org.awaitility.kotlin.await
+import org.junit.jupiter.api.Test
+import uk.gov.dluhc.printapi.config.IntegrationTest
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildCertificate
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildDelivery
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildPrintRequest
+import uk.gov.dluhc.printapi.testsupport.testdata.model.buildApplicationRemovedMessage
+import java.time.LocalDate
+import java.util.concurrent.TimeUnit
+
+internal class ApplicationRemovedMessageListenerTest : IntegrationTest() {
+
+    @Test
+    fun `should process message and set delivery info removal date`() {
+        // Given
+        val certificate = buildCertificate(
+            issueDate = LocalDate.of(2023, 4, 1), // to include 3 bank holidays in calculation
+            printRequests = listOf(
+                buildPrintRequest(delivery = buildDelivery()),
+                buildPrintRequest(delivery = buildDelivery())
+            )
+        )
+        certificateRepository.save(certificate)
+        val payload = buildApplicationRemovedMessage(
+            sourceReference = certificate.sourceReference!!,
+            gssCode = certificate.gssCode!!
+        )
+        val expectedInitialRemovalDate = LocalDate.of(2023, 5, 16)
+
+        // When
+        sqsMessagingTemplate.convertAndSend(applicationRemovedQueueName, payload)
+
+        // Then
+        await.atMost(5, TimeUnit.SECONDS).untilAsserted {
+            val response = certificateRepository.findAll()
+            assertThat(response).hasSize(1)
+            val saved = response[0]
+            assertThat(saved.initialRetentionRemovalDate).isEqualTo(expectedInitialRemovalDate)
```

Comment:
Minor, we already have a `CertificateAssert` class (`uk.gov.dluhc.printapi.testsupport.assertj.assertions.CertificateAssert`), so perhaps you could add a `hasInitialRetentionRemovalDate` method on it and use that?

Reply:
Done, but to be honest it felt a bit overkill at this stage since a) this is currently only used in one place and b) I had to add this sizeable method to our Assertions class:

```
    /**
     * Verifies that the actual Certificate's initialRetentionRemovalDate is equal to the given one.
     * @param initialRetentionRemovalDate the given initialRetentionRemovalDate to compare the actual Certificate's one.
     * @return this assertion object.
     * @throws AssertionError - if the actual Certificate's initialRetentionRemovalDate is not equal to the given one.
     */
    fun hasInitialRetentionRemovalDate(initialRetentionRemovalDate: LocalDate?): CertificateAssert {
        // check that actual Certificate we want to make assertions on is not null.
        isNotNull

        // overrides the default error message with a more explicit one
        val assertjErrorMessage = "\nExpecting initialRetentionRemovalDate of:\n  <%s>\nto be:\n  <%s>\nbut was:\n  <%s>"

        // null safe check
        val actualInitialRetentionRemovalDate = actual!!.initialRetentionRemovalDate
        if (!Objects.deepEquals(actualInitialRetentionRemovalDate, initialRetentionRemovalDate)) {
            failWithMessage(assertjErrorMessage, actual, initialRetentionRemovalDate, actualInitialRetentionRemovalDate)
        }

        // return the current assertion for method chaining
        return this
    }
```

That's not to say I'm not a fan of these custom assertion methods btw - I think they read really nicely - just that I wouldn't normally add them for one off use cases. If we start using it in more places though then it's definitely worth it.